ProjectFiles/Assets/Scripts/Game/Entity/Player.cs

```diff
@@ -0,0 +1,160 @@
+using System;
+using System.Runtime.Serialization;
+using Game.Core.Driving;
+using Network;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace Game.Entity {
+    /// <summary>
+    /// Used for storing all details about a player
+    /// </summary>
+    public class Player {
+        // Public Fields
+        public int           PlayerID      { get; }
+        public bool          IsEliminated  { get; set; }
+        public PlayerOptions PlayerOptions => playerOptions;
+
+        // Private Fields
+        private GameObject    car; // Do not directly set this. Use SetCar method
+        private Rigidbody     carRb;
+        private Transform     carTrans;
+        private bool          isSpawned;
+        private PlayerOptions playerOptions;
+        
+        private readonly bool isMe;
+
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="playerID">The network ID for the player</param>
+        /// <param name="options">The options used to configure the player</param>
+        /// <param name="me">Optional. Used to set the player to controllable.</param>
+        public Player(int playerID, PlayerOptions options, bool me = false) {
+            PlayerID      = playerID;
+            playerOptions = options;
+            isMe          = me;
+            if (isMe) {
+                ClientConfig.Me = this;
+            }
+        }
+
+        /// <summary>
+        /// Used to set the car object for a player.
+        ///
+        /// Used rather than setting to car directly to allow for the rigidbody to be extracted.
+        /// Helps to avoid 
+        /// </summary>
+        /// <param name="newCar"></param>
+        private void SetCar(GameObject newCar) {
+            car      = newCar;
+            carRb    = newCar.GetComponent<Rigidbody>();
+            carTrans = newCar.transform;
+        }
```

Comment:
Maybe have a property thing instead? Ask @liamdalg

Reply:
Cant do because I would need car to have a private getter and setter which isn't allowed using those